### ISSUE TITLE
cattle-node-agent daemonset created with "ClusterFirst" dnsPolicy even though it is running hostNetwork: true

### ISSUE THREAD

**Issue by Oats87** [rancher/rancher#16757]
**What kind of request is this (question/bug/enhancement/feature request):**
Question/Bug

**Steps to reproduce (least amount of steps as possible):**
Deploy Rancher Agents on downstream cluster

**Result:**
cattle-node-agent daemonset is created with dnsPolicy as
`dnsPolicy: ClusterFirst`

**Question**
Is there a reason why we are not setting the dnsPolicy on the node-agent daemonset to 
`dnsPolicy: ClusterFirstWithHostNet`? This would allow users who have configured kube-dns to resolve the proper DNS entry for Rancher to establish connections with Rancher rather than having to configure the underlying hosts to resolve the Rancher instance (for example, if you are in public cloud with no control due to corporate policies over DNS servers, etc)

More information about this can be found here: https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy

One potential downside I can see from this is in the event that `kube-dns` dies, none of the agents will be able to connect to the Rancher cluster. 

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.x
- Installation option (single install/HA): any

**Comment by MaxDiOrio** [comment]
Yes.  We have the issue using Rancher with GKE.  Even though we have a stubDomain set up in Kube-dns, we still need to manually edit the resolved.conf file on each K8S node to point back to our Internal DNS servers.
